src/compiler/checker.ts

```diff
@@ -3062,6 +3064,26 @@ namespace ts {
             setObjectTypeMembers(type, emptySymbols, callSignatures, constructSignatures, stringIndexType, numberIndexType);
         }
 
+        function intersectTypes(type1: Type, type2: Type): Type {
+            return !type1 ? type2 : !type2 ? type1 : getIntersectionType([type1, type2]);

```

Comment:
I'm going to keep it the way it is, making it longer doesn't really make it any clearer.